src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -3,23 +3,24 @@
 import java.util.Random;
 
 public class FigureSupplier {
-    private static final int FIGURE_LENGTH = Figures.values().length;
+    private static final int FIGURE_LENGTH = 5;
     private static final Random random = new Random();
     private final ColorSupplier supplier = new ColorSupplier();
     private final int randomSide = random.nextInt(100);
 
     public Figure getRandomFigure() {
         int index = random.nextInt(FIGURE_LENGTH);
-        if (Figures.values()[index].equals(Figures.CIRCLE)) {
-            return new Circle(randomSide, supplier.getRandomColor());
-        } else if (Figures.values()[index].equals(Figures.ISOSCELESTRAPEZOID)) {
-            return new IsoscelesTrapezoid(randomSide, randomSide, supplier.getRandomColor());
-        } else if (Figures.values()[index].equals(Figures.RIGHTTRIANGLE)) {
-            return new RightTriangle(randomSide, randomSide, supplier.getRandomColor());
-        } else if (Figures.values()[index].equals(Figures.RECTANGLE)) {
-            return new Rectangle(randomSide, randomSide, supplier.getRandomColor());
-        } else {
-            return new Square(randomSide, supplier.getRandomColor());
+        switch (index) {
+            case 0 :
+                return new Circle(randomSide, supplier.getRandomColor());
+            case 1 :
+                return new IsoscelesTrapezoid(randomSide, randomSide, supplier.getRandomColor());
+            case 2 :
+                return new RightTriangle(randomSide, randomSide, supplier.getRandomColor());
+            case 3 :
+                return new Rectangle(randomSide, randomSide, supplier.getRandomColor());

```

Comment:
randomSide - is always the same value. It is that behavior that we expect from the method with the name getRandomFigure. Let's generate a new value inside the method

Reply:
So your Rectangle will always have sides of the same length)
It is still not the expected behaviour